static/src/javascripts/projects/common/modules/article/progress.js

```diff
@@ -0,0 +1,36 @@
+define([
+    'common/utils/$',
+    'common/utils/mediator'
+], function (
+    $,
+    mediator
+) {
+    var articleHeight;
+
+    return {
+        init: function () {
+            // Check if progress bar is present
+            if ($('.progress').length) {
+                this.getArticleHeight();
+                this.bindEvents();
+            }
+        },
+
+        bindEvents: function () {
+            mediator.on('window:throttledScroll', this.updateProgress.bind(this));
+            mediator.on('window:resize', this.getArticleHeight.bind(this));
+        },
+
+        updateProgress: function () {
+            $('.progress__indicator').css('width', this.getProgressAsPercentage);
```

Comment:
If you had these as just private functions, you wouldn't need to do any `this` binding malarkey.